arangod/IResearch/IResearchView.cpp

```diff
@@ -297,18 +282,21 @@ struct IResearchView::ViewFactory : public arangodb::ViewFactory {
     //       for cluster the shards to lock come from coordinator and are not in
     //       the definition
     for (auto cid : metaState._collections) {
-      auto collection = vocbase.lookupCollection(
-          cid);  // always look up in vocbase (single server or cluster
-                 // per-shard collection)
-      auto link = collection
-                      ? IResearchLinkHelper::find(*collection, *impl)
-                      : nullptr;  // add placeholders to links, when the
-                                  // collection comes up it'll bring up the link
-
-      impl->_links.try_emplace(
-          cid, link ? link->self()
-                    : nullptr);  // add placeholders to links, when the link
-                                 // comes up it'll call link(...)
+      auto collection = vocbase.lookupCollection(cid);
+      // always look up in vocbase
+      // (single server or cluster per-shard collection)
+      auto link =
+          collection ? IResearchLinkHelper::find(*collection, *impl) : nullptr;
+      // add placeholders to links,
+      // when the collection comes up it'll bring up the link
+      auto [it1, was1] =
+          impl->_collections.try_emplace(cid, impl->_links.end());
```

Comment:
_collections same as old _links
_links only for collection link that is not null